homeassistant/components/android_ip_webcam/sensor.py

```diff
@@ -1,75 +1,182 @@
+)
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.icon import icon_for_battery_level
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-from . import (
-    CONF_HOST,
-    CONF_NAME,
-    CONF_SENSORS,
-    DATA_IP_WEBCAM,
-    ICON_MAP,
-    KEY_MAP,
-    AndroidIPCamEntity,
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .coordinator import AndroidIPCamDataUpdateCoordinator
+from .entity import AndroidIPCamBaseEntity
+
+
+@dataclass
+class AndroidIPWebcamSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[PyDroidIPCam], StateType]
+
+
+@dataclass
+class AndroidIPWebcamSensorEntityDescription(
+    SensorEntityDescription, AndroidIPWebcamSensorEntityDescriptionMixin
+):
+    """Entity description class for Android IP Webcam sensors."""
+
+    unit_fn: Callable[[PyDroidIPCam], str | None] = lambda _: None
+
+
+SENSOR_TYPES: tuple[AndroidIPWebcamSensorEntityDescription, ...] = (
+    AndroidIPWebcamSensorEntityDescription(
+        key="audio_connections",
+        name="Audio Connections",
+        icon="mdi:speaker",
+        state_class=SensorStateClass.TOTAL,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda ipcam: ipcam.status_data.get("audio_connections"),
+    ),
+    AndroidIPWebcamSensorEntityDescription(
+        key="battery_level",
+        name="Battery Level",
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda ipcam: ipcam.export_sensor("battery_level")[0],
+        unit_fn=lambda ipcam: ipcam.export_sensor("battery_level")[1],
+    ),
+    AndroidIPWebcamSensorEntityDescription(
+        key="battery_temp",
+        name="Battery Temperature",
+        icon="mdi:thermometer",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda ipcam: ipcam.export_sensor("battery_temp")[0],
+        unit_fn=lambda ipcam: ipcam.export_sensor("battery_temp")[1],
+    ),
+    AndroidIPWebcamSensorEntityDescription(
+        key="battery_voltage",
+        name="Battery Voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda ipcam: ipcam.export_sensor("battery_voltage")[0],
+        unit_fn=lambda ipcam: ipcam.export_sensor("battery_voltage")[1],
+    ),
+    AndroidIPWebcamSensorEntityDescription(
+        key="light",
+        name="Light Level",
+        icon="mdi:flashlight",
+        device_class=SensorDeviceClass.ILLUMINANCE,
```

Comment:
We shouldn't set device class until we have mapped the descriptions to the library units and added the Home Assistant units to the descriptions. We require specific units when setting a device class.

Let's do this in the next PR. Please remove the sensor device classes for now.

Reply:
Here's an example how to map device units to sensor entity descriptions:
https://github.com/home-assistant/core/blob/7d427ddbd4b5eca09b0dfa6f7523e2efd2af5afc/homeassistant/components/xiaomi_ble/sensor.py#L41-L47

`DeviceClass` and `Units` are 3rd party library classes.